scripts/migrations/update_domain_codes.py

```diff
@@ -0,0 +1,79 @@
+'''
+update_domain_codes.py
+
+A module that overwrites all of the domain codes with their associated descriptions.
+It also adjusts field lengths if needed.
+'''
+
+from copy import copy
+from json import dumps
+from os.path import dirname, join
+from textwrap import dedent
+
+import arcpy
+
+arcpy.env.workspace = sde = join(dirname(__file__), 'local@electrofishing.sde')
+datasets = arcpy.ListTables() + arcpy.ListFeatureClasses()
+
+domains_in_data = {}
+
+
+print('finding domains and their relationship to data')
+for dataset in datasets:
+    print('\n' + dataset)
+    for field in arcpy.ListFields(dataset):
+        if len(field.domain) == 0 or field.type != 'String':
+            continue
+
+        print(field.name)
+
+        domains_in_data.setdefault(field.domain, []).append((dataset, field))
+
+#: look for orphaned domains
+all_domains = [domain for domain in arcpy.da.ListDomains() if domain.domainType == 'CodedValue']
+all_domain_names = set(domain.name for domain in all_domains)
+used_domains = set(domains_in_data.keys())
+
+
+orphans = all_domain_names - used_domains
+
+if len(orphans) > 0:
+    print('orphans: {}'.format(orphans))
```

Comment:
does orphans print right or does it need to be `','.join()`

Reply:
The join is not needed. I believe that when you pass a non-string into `.format(...)` it calls str() or something similar.

This actually doesn't print anything on the current database. I just wanted to be sure that we didn't have any.